File: 217_contains_duplicate/contains_duplicate.py
from typing import List

class Solution:
    def contains_duplicate(self, nums: List[int]) ->bool:
        hashset = set()
        for n in nums:
            if n in hashset:
                return True
        
            hashset.add(n)
        
        return False


if __name__ == "__main__":
    solution = Solution()
    nums = [1,2,3,1]
    result = solution.contains_duplicate(nums)
    print(result)

File: 217_contains_duplicate/contains_duplicate.go
package main

import "fmt"

type Solution struct {
}

func (*Solution) containsDuplicate(nums []int) bool {
	seen := make(map[int]bool)

	for _, n := range nums {
		if seen[n] {
			return true
		}
		seen[n] = true
	}

	return false
}

func main() {
	solution := new(Solution)
	nums := []int{1, 2, 3, 1}
	result := solution.containsDuplicate(nums)
	fmt.Println(result)

}
